apps/dav/lib/Connector/Sabre/CommentPropertiesPlugin.php

```diff
@@ -115,29 +116,27 @@ public function handleGetProperties(
 			return $this->commentsManager->getNumberOfCommentsForObject('files', (string)$node->getId());
 		});
 
-		$propFind->handle(self::PROPERTY_NAME_HREF, function () use ($node) {
+		$propFind->handle(self::PROPERTY_NAME_HREF, function () use ($node): ?string {
 			return $this->getCommentsLink($node);
 		});
 
-		$propFind->handle(self::PROPERTY_NAME_UNREAD, function () use ($node): ?bool {
+		$propFind->handle(self::PROPERTY_NAME_UNREAD, function () use ($node): ?int {
 			return $this->cachedUnreadCount[$node->getId()] ?? $this->getUnreadCount($node);
 		});
 	}
 
 	/**
 	 * Returns a reference to the comments node
-	 *
-	 * @return array|string|null
 	 */
-	public function getCommentsLink(Node $node) {
+	public function getCommentsLink(Node $node): ?string {
 		$href = $this->server->getBaseUri();
 		$entryPoint = strpos($href, '/remote.php/');
 		if ($entryPoint === false) {
 			// in case we end up somewhere else, unexpectedly.
 			return null;
 		}
-		$commentsPart = 'dav/comments/files/' . rawurldecode($node->getId());
-		$href = substr_replace($href, $commentsPart, $entryPoint + strlen('/remote.php/'));
+		$commentsPart = 'dav/comments/files/' . rawurldecode((string)$node->getId());
+		$href = (string)substr_replace($href, $commentsPart, $entryPoint + strlen('/remote.php/'));
```

Comment:
RedundantCast: Redundant cast to string
```suggestion
		$href = substr_replace($href, $commentsPart, $entryPoint + strlen('/remote.php/'));
```